resilience4j-retry/src/main/java/io/github/resilience4j/retry/event/AbstractRetryEvent.java

```diff
@@ -18,16 +18,19 @@
  */
 package io.github.resilience4j.retry.event;
 
+import io.github.resilience4j.core.lang.Nullable;
+
 import java.time.ZonedDateTime;
 
 abstract class AbstractRetryEvent implements RetryEvent {
 
     private final String name;
     private final ZonedDateTime creationTime;
     private final int numberOfAttempts;
+    @Nullable
     private final Throwable lastThrowable;
 
-    AbstractRetryEvent(String name, int numberOfAttempts, Throwable lastThrowable) {
+    AbstractRetryEvent(String name, int numberOfAttempts, @Nullable Throwable lastThrowable) {
```

Comment:
We have one instance in the code that passes a null into the `lastThrowable` value. So that forced this to be nullable. which means the `toString` implementation of its children also need to be updated to handle that. Should I just have the `toString` default to `null` when it isn't set?